src/Gating/GateDataSet.cs

```diff
@@ -1781,6 +1811,38 @@ private static void ReadEndDate(GatingConfiguration.BaseGateType configurationGa
 		}
 
 
+		/// <summary>
+		/// Read the cloud contexts from a configuration gate
+		/// </summary>
+		/// <param name="configurationGate">configuration to read from</param>
+		/// <param name="gate">gate to add information to</param>
+		private void ReadCloudContexts(GatingConfiguration.BaseGateType configurationGate, Gate gate)
+		{
+			if (configurationGate.CloudContexts == null)
+			{
+				return;
+			}
+
+			gate.CloudContexts = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+			foreach (GatingConfiguration.BaseGateTypeCloudContext context in configurationGate.CloudContexts)
+			{
+				if (context.Name == "None")
```

Comment:
>context.Name  [](start = 8, length = 13)

Are we sure at this point that context.Name is not null?

Reply:
I copied the behaviour of ReadHostEnvironments

Reply:
Yes, I think we can, because without the property Name context doesn't pass through XML deserialization - the whole Gate would be read as `null`.
In unit tests this results in an assertion failure.